### DISCUSSION TITLE
Use Renovate to update apache/maven

### DISCUSSION THREAD

**Discussion by y0x** [renovatebot/renovate#32250] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I´m building a Docker image where I install maven. I want renovate to update my ENV variable based on github-tags. Currently I´m only able to get this working when I define the variable like this:

```
# renovate: datasource=github-tags depName=apache/maven
ENV MAVEN_VERSION=maven-3.9.5
```

This does not work:

```
# renovate: datasource=github-tags depName=apache/maven
ENV MAVEN_VERSION=3.9.5
```

During the installation process I´m required at one point to have the maven version variable without the "maven-", so like this `ENV MAVEN_VERSION=3.9.5`. The first $MAVEN_VERSION in the following string need just 3.9.5

`
https://archive.apache.org/dist/maven/maven-3/$MAVEN_VERSION/binaries/apache-$MAVEN_VERSION-bin.tar.gz
`

Is there a way to automatically let renovate add maven-{version} within the config or what other options could I have to solve this?

```
# renovate: datasource=github-tags depName=apache/maven
ENV MAVEN_VERSION=maven-3.9.5 

RUN set -ex \
    # Install Maven
    && mkdir -p $MAVEN_HOME \
    && curl -LSso /var/tmp/apache-$MAVEN_VERSION-bin.tar.gz https://archive.apache.org/dist/maven/maven-3/$MAVEN_VERSION/binaries/apache-$MAVEN_VERSION-bin.tar.gz \
    && tar xzf /var/tmp/apache-$MAVEN_VERSION-bin.tar.gz -C $MAVEN_HOME --strip-components=1 \
    && rm /var/tmp/apache-$MAVEN_VERSION-bin.tar.gz \
    && update-alternatives --install /usr/bin/mvn mvn /opt/maven/bin/mvn 10000 \
    && mkdir -p $MAVEN_CONFIG_HOME 
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
You can, but you'll need to customize the regex which is used. 

You *might* be able to alternatively use versionCompatibility with the maven- part being an optional compatibility prefix

**Reply by y0x** [reply]
With this packageRule I was able to let renovate find the version on github

```
{
        "description": "maven update",
        "matchPackageNames": ["apache/maven"],
        "versioning":"regex:(maven-)?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?",
        //"extractVersion": "^(?<version>maven-\\d+\\.\\d+\\.\\d+)$",
        //"versionCompatibility": "^(?<compatibility>.*)-(?<version>.*)$",
    }
```

The problem now is that in the merge request Renovate adds the maven- again:

```
# renovate: datasource=github-tags depName=apache/maven versioning=loose
- ENV MAVEN_VERSION=3.9.5
+ ENV MAVEN_VERSION=maven-3.9.9
```

For now I was not able to get it working with versionCompatibility. Do you have any example snippet on how to do this for my usecase?
